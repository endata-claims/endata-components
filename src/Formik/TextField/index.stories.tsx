import React from 'react'
import TextField from '.'
import { Formik, Form } from 'formik'

export default {
  title: 'TextField'
}

export const Default = () => (
  <Formik
    initialValues={{ text: '' }}
    onSubmit={(values, actions) => {
      console.log(values)
      actions.setSubmitting(false)
    }}
  >
    <Form>
      <TextField label='Text field' name='text' size='small' />
    </Form>
  </Formik>
)
