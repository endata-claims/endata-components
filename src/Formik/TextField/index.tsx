import React from 'react'

import { TextField, TextFieldProps } from 'formik-material-ui'
import { InputAdornment } from '@material-ui/core'

export type FormikTextFieldProps = TextFieldProps & {
  startAdornment?: React.ReactElement
  endAdornment?: React.ReactElement
  readOnly?: boolean
}
const FormikTextField: React.FC<FormikTextFieldProps> = ({ startAdornment, endAdornment, InputProps, readOnly = false, ...props }) => {
  return (
    <TextField
      {...props}
      InputProps={{
        ...InputProps,
        startAdornment: !startAdornment ? null : (
          <InputAdornment position='start'>
            {startAdornment}
          </InputAdornment>
        ),
        endAdornment: !endAdornment ? null : (
          <InputAdornment position='end'>
            {endAdornment}
          </InputAdornment>
        ),
        readOnly
      }}
    />
  )
}
export default FormikTextField