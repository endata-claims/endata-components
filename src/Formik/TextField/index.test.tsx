import React from 'react'
import { shallow } from 'enzyme'

import TextField from '.'
import { Formik, Form } from 'formik'

describe('Test Button', () => {
  const renderWrapper = () => shallow(
    <Formik
      initialValues={{ text: 'dkm' }}
      onSubmit={values => console.log(values)}
    >
      <Form>
        <TextField label='Text Field' name='text' />
      </Form>
    </Formik>
  )

  describe('Snapshots', () => {
    it('should match snapshots with label', () => {
      expect(renderWrapper()).toMatchSnapshot()
    })
  })
})
