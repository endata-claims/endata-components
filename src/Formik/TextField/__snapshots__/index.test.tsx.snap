// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Snapshots should match snapshots with label 1`] = `
<ContextProvider
  value={
    Object {
      "dirty": false,
      "errors": Object {},
      "getFieldHelpers": [Function],
      "getFieldMeta": [Function],
      "getFieldProps": [Function],
      "handleBlur": [Function],
      "handleChange": [Function],
      "handleReset": [Function],
      "handleSubmit": [Function],
      "initialErrors": Object {},
      "initialStatus": undefined,
      "initialTouched": Object {},
      "initialValues": Object {
        "text": "dkm",
      },
      "isSubmitting": false,
      "isValid": true,
      "isValidating": false,
      "registerField": [Function],
      "resetForm": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFormikState": [Function],
      "setStatus": [Function],
      "setSubmitting": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "status": undefined,
      "submitCount": 0,
      "submitForm": [Function],
      "touched": Object {},
      "unregisterField": [Function],
      "validateField": [Function],
      "validateForm": [Function],
      "validateOnBlur": true,
      "validateOnChange": true,
      "validateOnMount": false,
      "values": Object {
        "text": "dkm",
      },
    }
  }
>
  <Form>
    <FormikTextField
      label="Text Field"
      name="text"
    />
  </Form>
</ContextProvider>
`;
